### PR TITLE
MM-17455 Added test for special keys that shouldn't put focus on main input

### PR DESCRIPTION
**Pull request by valentijnnieman**
#### Summary

This PR adds a test that tests if a bunch of special keys do not put focus back on the main input. Let me know if I'm missing anything! Thanks :-)

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/11752

### CODE REVIEW

File: e2e/cypress/integration/messaging/focus_move_spec.js [hunk 1]
```diff
@@ -119,4 +119,37 @@ describe('Messaging', () => {
             });
         });
     });
+    it('M17455 - Focus does not move for non-character keys', () => {
```
Comment (saturninoabril) [inline e2e/cypress/integration/messaging/focus_move_spec.js]: This is failing due to the opened "Add new members modal" from previous test.  Should do ``cy.visit('/')`` first to start the test (or better do ``cy.visit('/')`` before each test)

File: e2e/cypress/integration/messaging/focus_move_spec.js [hunk 2]
```diff
@@ -119,4 +119,37 @@ describe('Messaging', () => {
...
+        //# Make sure main input is visible and focused
+        cy.get('#post_textbox').should('be.visible');
+        cy.get('#post_textbox').should('be.focused');
```
Comment (saturninoabril) [inline e2e/cypress/integration/messaging/focus_move_spec.js]: Please combine:
```javascript
cy.get('#post_textbox').should('be.visible').and('be.focused');
```

File: e2e/cypress/integration/messaging/focus_move_spec.js [hunk 3]
```diff
@@ -119,4 +119,37 @@ describe('Messaging', () => {
...
+        //#Click anywhere in the body to move the focus out of the main input box
+        cy.get('body').click();
+
+        //# Make sure main input is not focused
+        cy.get('#post_textbox').should('not.be.focused');
+
+        // Keycodes for keys that don't have a special character sequence for cypress.type()
+        const numLockKeycode = 144;
+        const f7Keycode = 118;
+        const windowsKeycode = 91;
```
[the remark this answers is not in the document]
Comment (saturninoabril) [reply e2e/cypress/integration/messaging/focus_move_spec.js]: I'm okay to remain as is until we have another use for it.

File: e2e/cypress/integration/messaging/focus_move_spec.js (hunk 1 above)
Comment (saturninoabril) [reply e2e/cypress/integration/messaging/focus_move_spec.js]: Yes, failing on me.  Like I mentioned, the "Add new members modal" remains open before this test.

Comment (valentijnnieman) [reply e2e/cypress/integration/messaging/focus_move_spec.js]: Ah! I got confused between the `before()` hook and the `beforeEach()` hook :-) I think this should be good now - thanks for explaining!

**Comment by valentijnnieman** [comment]
Just to let you know, I'm getting this screen when trying to restart the CI workflow: 
Screenshot from 2019-09-09 14-40-05

**Comment by hanzei** [comment]
@valentijn You have to log into Circelci to get the build working. I restarted the build to fix this.

**Comment by valentijnnieman** [comment]
@hanzei Thanks for your response! I was logged in I believe, still got that screen.

**Pull request merged**